src/restconfig/src/main/java/org/geoserver/rest/resources/ResourceResource.java

```diff
@@ -0,0 +1,409 @@
+import org.geoserver.rest.RestletException;
+import org.geoserver.rest.format.DataFormat;
+import org.geoserver.rest.format.ReflectiveHTMLFormat;
+import org.geoserver.rest.format.ReflectiveJSONFormat;
+import org.geoserver.rest.format.ReflectiveXMLFormat;
+import org.geoserver.rest.util.IOUtils;
+import org.geoserver.rest.util.RESTUtils;
+import org.restlet.Context;
+import org.restlet.data.MediaType;
+import org.restlet.data.Method;
+import org.restlet.data.Request;
+import org.restlet.data.Response;
+import org.restlet.data.Status;
+import org.restlet.resource.InputRepresentation;
+import org.restlet.resource.Representation;
+
+import com.thoughtworks.xstream.annotations.XStreamAlias;
+
+import freemarker.ext.beans.CollectionModel;
+import freemarker.template.Configuration;
+import freemarker.template.SimpleHash;
+
+/**
+ *
+ *
+ * @author Niels Charlier
+ *
+ */
+public class ResourceResource extends AbstractResource {
+    
+    public enum Operation {DEFAULT, METADATA, MOVE, COPY};
+           
+    /**
+     *
+     * XML/Json object for resource reference.
+     *
+     */
+    protected static class ResourceReference {
+                
+        private String path;
+
+        private AtomLink link;
+        
+        public ResourceReference(String path, AtomLink link) {
+            this.path = path;
+            this.link = link;
+        }
+
+        public String getPath() {
+            return path;
+        }
+
+        public AtomLink getLink() {
+            return link;
+        }
+    }
+    
+    /**
+     * 
+     * XML/Json object for resource child of directory.
+     * 
+     */
+    @XStreamAlias("child")
+    protected static class ResourceChild {
+                
+        private String name;
+
+        private AtomLink link;
+        
+        public ResourceChild(String name, AtomLink link) {
+            this.name = name;
+            this.link = link;
+        }
+
+        public String getName() {
+            return name;
+        }
+
+        public AtomLink getLink() {
+            return link;
+        }
+    }
+    
+    /**
+     *
+     * XML/Json object for resource metadata.
+     *
+     */
+    @XStreamAlias("ResourceMetadata")
+    protected static class ResourceMetadata {
+        
+        private String name;
+        private ResourceReference parent;
+        private Date lastModified;
+        private String type;
+               
+        public ResourceMetadata(String name, ResourceReference parent, Date lastModified, String type) {
+            this.name = name;
+            this.parent = parent;
+            this.lastModified = lastModified;
+            this.type = type;
+        }
+
+        /**
+         * Create from resource.
+         * The class must be static for serialization, but output is request dependent so passing on self.
+         */
+        protected ResourceMetadata(Resource resource, ResourceResource self, boolean isDir) {
+            if (!"".equals(resource.path())) {
```

Comment:
@NielsCharlier this is a strange idiom. Does your intent differ from ```!resource.path().isEmpty()```? Can ```resource.path()``` return null?